src/main/java/com/synopsys/arc/jenkinsci/plugins/extratoolinstallers/installers/RarExtractionInstaller.java

```diff
@@ -0,0 +1,157 @@
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+/**
+ * Installs a tool by downloading and unpacking a RAR file. 
+ * The installer is inspired by "ZipExtractionInstaller" from the Custom Tools plugin. 
+ * Limitation: Extracts only a single RAR file. A tool archive split into multiple RAR volumes will be not fully extracted.
+ *
+ * @author Martin Hjelmqvist <martin@hjelmqvist.eu>.
+ */
+public class RarExtractionInstaller extends AbstractExtraToolInstaller {
+
+    /**
+     * Messages displayed when configuring the tool in Jenkins. 
+     * TODO: Move this to Messages.java generated by the localizer.
+     */
+    private static final String RAR_EXTRACTION_INSTALLER_DISPLAY_NAME = "Extract *.rar";
+    private static final String RAR_EXTRACTION_INSTALLER_BAD_CONNECTION = "Server rejected connection.";
+    private static final String RAR_EXTRACTION_INSTALLER_MALFORMED_URL = "Malformed URL.";
+    private static final String RAR_EXTRACTION_INSTALLER_COULD_NOT_CONNECT = "Could not connect to URL.";
+    private static final String RAR_EXTRACTION_INSTALLER_NO_TOOLHOME = "Must provide a tool home directory.";
+
+    /**
+     * Constructs a RarExtractionInstaller.
+     *
+     * @param label Optional label for this installer.
+     * @param toolHome Path to the location of the tool archive.
+     * @param failOnSubstitution Fail the build if substitution fails.
+     */
+    @DataBoundConstructor
+    public RarExtractionInstaller(String label, String toolHome,
+            boolean failOnSubstitution) {
+        super(label, toolHome, failOnSubstitution);
+    }
+
+    /**
+     * Performs installation of tool on the node. 
+     * If the tool archive is e.g. corrupted, exception will be thrown from the JUnrar library.
+     */
+    @Override
+    public FilePath performInstallation(ToolInstallation tool, Node node, TaskListener log) throws IOException, InterruptedException {
+        FilePath dir = preferredLocation(tool, node);
+        RarFilePath rarDir = new RarFilePath(dir);
+        String toolHome = getToolHome();
+
+        if (rarDir.installIfNecessaryFrom(new URL(toolHome), log, "Unpacking " + toolHome + " to " + dir + " on " + node.getDisplayName())) {
+            rarDir.act(new ChmodRecAPlusX());
+        }
+        return dir;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends ToolInstallerDescriptor<RarExtractionInstaller> {
+
+        @Override
+        public String getDisplayName() {
+            return RAR_EXTRACTION_INSTALLER_DISPLAY_NAME;
+        }
+
+        public FormValidation doCheckUrl(@QueryParameter String value) {
+            try {
+                URLConnection con = ProxyConfiguration.open(new URL(value));
+                con.connect();
+                if (con instanceof HttpURLConnection) {
```

Comment:
Copied from existing code. Should this be different? Are they handling other connection types?